src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1963,7 +1955,9 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 float rv = ASR::down_cast<ASR::ConstantReal_t>(ASRUtils::expr_value(float_expr))->m_r;
                 tmp = ASR::make_ConstantReal_t(al, x.base.base.loc, rv, float_type);
             } else if (ASRUtils::is_integer(*float_type)) {
-                // convert an int to a float
+                // convert an int to a float using implicit cast
+                double rv = ASR::down_cast<ASR::ConstantInteger_t>(ASRUtils::expr_value(float_expr))->m_n;
+                tmp = ASR::make_ConstantReal_t(al, x.base.base.loc, rv, float_type);
```

Comment:
This is not calling the implicit cast ASR node.

We don't need to do it in this PR, but we need to do that. Then it will start working for runtime values too.